### ISSUE TITLE
Navigation Issue

### ISSUE THREAD

**Issue by bjmcallister** [DeepARSDK/deepar-flutter-plugin#21]
Have not tested on Android, but on iOS when I initialize the camera controller and then navigate to a page and back to the camera page. Black screen shows with zero errors.

Anyone else have this issue?

**Comment by bjmcallister** [comment]
Can confirm that this is also an issue on Android.

**Comment by bjmcallister** [comment]
On android this is spammed.

"E/BufferQueueProducer( 3365): [SurfaceTexture-0-3365-0](id:d2500000000,api:1,p:3365,c:3365) dequeueBuffer: BufferQueue has been abandoned
I/Adreno  ( 3365): DequeueBuffer: dequeueBuffer failed"

**Comment by J4ckSk3le7on** [comment]
Here the same, for the moment it is the only thing that is failing, hopefully they can fix it soon, they made a great package.

**Comment by wilson-christian** [comment]
@bjmcallister  @J4ckSk3le7on 
we tried running the `example` app in both android & iOS real devices which works fine. Can you provide minimum reproducible code please?

**Comment by bjmcallister** [comment]
Just use Navigator.push() to open a new page that contains the deepar example and the Navigator.pop() to go back to the previous page. Once there use Navigator.push() to go back to the page with the deepar example and it will no longer work.

**Comment by J4ckSk3le7on** [comment]
This is a simple executable, everything works correctly until you try to reopen the camera.

As a bonus, if you try to call the function 'deepController.destroy()' the application crashes and closes.

```
import 'package:deepar_flutter/deepar_flutter.dart';
import 'package:flutter/material.dart';
import 'package:flutter/scheduler.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: FirstPage()
    );
  }
}

class FirstPage extends StatefulWidget {
  const FirstPage({Key? key}) : super(key: key);

  @override
  State<FirstPage> createState() => _FirstPageState();
}

class _FirstPageState extends State<FirstPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('FirstPage'),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Text('Just go to screen two and come back, then try to open the camera again.'),
          ElevatedButton(
            onPressed: () {
              Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => const SecondPage())
              );
            },
            child: const Text('Open Camera')
          )
        ],
      ),
    );
  }
}

//////////////////////////////////////////////////////////////////////

class SecondPage extends StatefulWidget {
  const SecondPage({Key? key}) : super(key: key);

  @override
  State<SecondPage> createState() => _SecondPageState();
}

class _SecondPageState extends State<SecondPage> {
  DeepArController deepController = DeepArController();
  String filePath = '';

  @override
  void initState() {
    SchedulerBinding.instance.addPostFrameCallback((timeStamp) async {
      await initializeDeepController();
    });
    super.initState();
  }

  Future initializeDeepController() async {
    await deepController.initialize(
      androidLicenseKey: YOUR LICENSE,
      iosLicenseKey: YOUR LICENSE,
      resolution: Resolution.high,
    ).then((value) => null).whenComplete(() {
      setState(() {});
    });
  }

  @override
  void dispose() {
    //------> If you call this function, the application simply crashes.
    // deepController.destroy();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('SecondPage'),
      ),
      body: Center(
        child: Stack(
          children: [
            deepController.isInitialized == false ?
              const CircularProgressIndicator() :
                DeepArPreview(deepController),
            Align(
              alignment: Alignment.center,
              child: Text(filePath),
            )
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () async {
          if (deepController.isRecording) {
            await deepController.stopVideoRecording().then((value) {
              setState(() => filePath = value.path);
            });
          } else {
            await deepController.startVideoRecording();
            setState(() {});
          }
        },
        child: Icon(deepController.isRecording ? Icons.stop : Icons.play_arrow),
      ),
    );
  }
}
```

**Comment by bjmcallister** [comment]
Thank you @J4ckSk3le7on. I've been too busy to write this up. That is exactly the issue.

**Comment by IBRAHIMBNC** [comment]
> Have not tested on Android, but on iOS when I initialize the camera controller and then navigate to a page and back to the camera page. Black screen shows with zero errors.
> 
> Anyone else have this issue?

I am facing the same issue

**Comment by wilson-christian** [comment]
We have fixed this in the latest release 0.0.2
